pkg/teleporter/relayer.go

```diff
@@ -267,71 +297,60 @@ func getRelayerURL(version string) (string, error) {
 	), nil
 }
 
-func UpdateRelayerConfig(
-	relayerConfigPath string,
-	relayerStorageDir string,
-	relayerAddress string,
-	relayerPrivateKey string,
-	network models.Network,
-	subnetID string,
-	blockchainID string,
-	teleporterContractAddress string,
-	teleporterRegistryAddress string,
-) error {
+func loadRelayerConfig(relayerConfigPath string) (*config.Config, error) {
 	awmRelayerConfig := config.Config{}
-	if utils.FileExists(relayerConfigPath) {
-		bs, err := os.ReadFile(relayerConfigPath)
-		if err != nil {
-			return err
-		}
-		if err := json.Unmarshal(bs, &awmRelayerConfig); err != nil {
-			return err
-		}
-	} else {
-		awmRelayerConfig = createRelayerConfig(
-			logging.Info.LowerString(),
-			relayerStorageDir,
-			network.Endpoint,
-		)
-	}
-	host, port, _, err := utils.GetURIHostPortAndPath(network.Endpoint)
+	bs, err := os.ReadFile(relayerConfigPath)
 	if err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(bs, &awmRelayerConfig); err != nil {
+		return nil, err
+	}
+	return &awmRelayerConfig, nil
+}
+
+func saveRelayerConfig(relayerConfig *config.Config, relayerConfigPath string) error {
+	if err := os.MkdirAll(filepath.Dir(relayerConfigPath), constants.DefaultPerms755); err != nil {
 		return err
 	}
-	addChainToRelayerConfig(
-		&awmRelayerConfig,
-		host,
-		port,
-		subnetID,
-		blockchainID,
-		teleporterContractAddress,
-		teleporterRegistryAddress,
-		relayerAddress,
-		relayerPrivateKey,
-	)
-	bs, err := json.MarshalIndent(awmRelayerConfig, "", "  ")
+	bs, err := json.MarshalIndent(relayerConfig, "", "  ")
 	if err != nil {
 		return err
 	}
-	if err := os.WriteFile(relayerConfigPath, bs, constants.WriteReadReadPerms); err != nil {
-		return err
+	return os.WriteFile(relayerConfigPath, bs, constants.WriteReadReadPerms)
+}
+
+func CreateBaseRelayerConfigIfMissing(
```

Comment:
nit: I am not sure if we need this func, no reason to abstract `!utils.FileExists(relayerConfigPath)` logic. 
pls ignore this, no strong opinion

Reply:
ok keeping it for the moment